impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/mvnup/UpgradeContext.java

```diff
@@ -159,4 +167,27 @@ public void action(String message) {
     public UpgradeOptions options() {
         return (UpgradeOptions) super.options();
     }
+
+    /**
+     * Detects if the current console supports Unicode characters.
+     * Uses the terminal's stdout encoding to determine Unicode support.
+     *
+     * @return true if Unicode is likely supported, false otherwise
+     */
+    private boolean supportsUnicode() {
+        try {
+            // Use the terminal's actual stdout encoding if available
+            if (terminal != null && terminal.stdoutEncoding() != null) {
+                String encoding = terminal.stdoutEncoding().name().toLowerCase();
+                // UTF-8 and UTF-16 encodings support Unicode
+                return encoding.contains("utf");
+            }
+        } catch (Exception e) {
```

Comment:
Please be specific about exceptions

Reply:
'm with elharo. Please use specific exception handling